### DISCUSSION TITLE
How can I determine the Big O complexity of this algorithm?

### DISCUSSION THREAD

**Discussion by Samuel-5485** [community/community#137316] [Code Security] [answered]
### Select Topic Area

Question

### Body

I'm trying to understand how to calculate the Big O complexity of an algorithm. I understand the basics, but I'm unsure how to apply it to a specific piece of logic.

For example, if I have a nested loop where the outer loop runs n times and the inner loop runs n/2 times, how do I determine the overall Big O complexity?

**### Question!** 
Can someone guide me on how to analyze the time complexity in this scenario and explain how to correctly express it in Big O notation? Any tips on how to approach these kinds of problems would also be appreciated!

**Comment by Mrsanyi123** [comment, marked as answer]
The overall time complexity of the algorithm is O(n^2).
